app/src/main/java/ru/thstdio/aa2020/api/Repository.kt

```diff
@@ -1,51 +1,68 @@
-import kotlinx.serialization.json.Json
-import ru.thstdio.aa2020.api.response.*
-import ru.thstdio.aa2020.data.Actor
-import ru.thstdio.aa2020.data.Genre
-import ru.thstdio.aa2020.data.Movie
-
-class Repository(private val appContext: Context) {
-    private val jsonFormat = Json { ignoreUnknownKeys = true }
-
-    suspend fun downloadMovies(): List<Movie> {
-        val moviesJson = loadFromJson<List<JsonMovie>>("data.json", Json::decodeFromString)
-
-        val actors = loadActors().associateBy { it.id }
-        val genres = loadGenres().associateBy { it.id }
-        return moviesJson.map { it.toMovie(actors, genres) }
-    }
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.serialization.ExperimentalSerializationApi
+import ru.thstdio.aa2020.api.converter.toActor
+import ru.thstdio.aa2020.api.converter.toCinema
+import ru.thstdio.aa2020.api.converter.toCinemaDetail
+import ru.thstdio.aa2020.api.response.ConfigurationResponse
+import ru.thstdio.aa2020.api.response.Genre
+import ru.thstdio.aa2020.api.service.Service
+import ru.thstdio.aa2020.api.service.TimeDbApi
+import ru.thstdio.aa2020.data.CinemaDetail
+import ru.thstdio.aa2020.data.CinemaListWithTotalPage
 
-    suspend fun downloadMovie(id: Long): Movie =
-        withContext(Dispatchers.IO) {
-            delay(1400)
-            downloadMovies().first { it.id == id }
-        }
 
+@ExperimentalSerializationApi
+class Repository() {
+    private val api: TimeDbApi = Service().createApi()
 
-    private suspend fun loadGenres(): List<Genre> =
-        loadFromJson<List<JsonGenre>>("genres.json", Json::decodeFromString)
-            .map { it.toGenre() }
+    private var configuration: ConfigurationResponse? = null
+    private var genres: Map<Long, Genre>? = null
 
-    private suspend fun loadActors(): List<Actor> =
-        loadFromJson<List<JsonActor>>("people.json", Json::decodeFromString)
-            .map { it.toActor() }
 
+    suspend fun getMoviesFromPage(page: Int): CinemaListWithTotalPage = coroutineScope {
+        val configurationAndGenres = async { getConfigurationAndGenres() }
+        val responseAsync = async { api.getNowPlaying(page) }
+        val (configuration, genres) = configurationAndGenres.await()
+        val response = responseAsync.await()
+        CinemaListWithTotalPage(
+            list = response.results.map { item ->
+                item.toCinema(
+                    configuration,
+                    genres
+                )
+            },
+            totalPage = response.totalPages
+        )
+    }
 
-    private suspend fun <T> loadFromJson(fileName: String, parse: Json.(String) -> T): T =
-        withContext(Dispatchers.IO) {
-            val jsonString = readAssetFileToString(fileName)
-            jsonFormat.parse(jsonString)
+    private suspend fun getConfigurationAndGenres(): Pair<ConfigurationResponse, Map<Long, Genre>> =
+        coroutineScope {
+            val configurationAsync = async {
```

Comment:
val configurationAsync = async {
    val alreadyLoadedConfiguration = configurationAtomic.get()
    if(alreadyLoadedConfiguration!= null) {
        alreadyLoadedConfiguration
    } else {
        val justLoadedConfiguration = api.getConfiguration()
        configurationAtomic.set(justLoadedConfiguration)
       justLoadedConfiguration
    }
}

and this part looks like new private fun(except async{})